tests/services/test_job_execution.py

```diff
@@ -535,3 +535,30 @@ def test_load_max_concurrent_access_control_syncs_from_config():
 )
 def test_load_max_concurrent_access_control_syncs_fallback_on_default():
     assert load_max_concurrent_access_control_syncs({}) == MAX_SIX_CONCURRENT_SYNCS
+
+
+def test_override_es_config():
+    config = {
+        "elasticsearch": {
+            "username": "username",
+            "password": "password",
+            "api_key": "global_api_key",
+        },
+        "service": {"idling": 30},
+        "sources": [],
+        "connectors": [
+            {
+                "connector_id": "foo",
+                "service_type": "bar",
+                "api_key": "connector_api_key",
+            }
+        ],
+    }
+
+    service = JobExecutionService(config)
+    connector = Mock()
+    connector.id = "foo"
+    override_config = service._override_es_config(connector)
+    assert "username" not in override_config
+    assert "password" not in override_config
+    assert override_config["api_key"] == "connector_api_key"
```

Comment:
```suggestion
   connector_api_key = "connector_api_key"

    config = {
        "elasticsearch": {
            "username": "username",
            "password": "password",
            "api_key": "global_api_key",
        },
        "service": {"idling": 30},
        "sources": [],
        "connectors": [
            {
                "connector_id": "foo",
                "service_type": "bar",
                "api_key": connector_api_key,
            }
        ],
    }

    service = JobExecutionService(config)
    connector = Mock()
    connector.id = "foo"
    override_config = service._override_es_config(connector)
    assert "username" not in override_config
    assert "password" not in override_config
    assert override_config["api_key"] == connector_api_key
```

nit: we could use a variable (`connector_api_key`) here